packages/gatsby/cache-dir/production-app.js

```diff
@@ -80,33 +81,30 @@ apiRunnerAsync(`onClientEntry`).then(() => {
     }
   }
 
+  const { page, location: browserLoc } = window
   if (
-    window.page &&
-    window.page.path !== `/404.html` &&
-    !window.page.path.match(/^\/offline-plugin-app-shell-fallback\/?$/) &&
-    __PATH_PREFIX__ + window.page.path !== window.location.pathname
+    page &&
+    page.path !== `/404.html` &&
+    (!page.matchPath || !match(page.matchPath, browserLoc.pathname)) &&
```

Comment:
should `__PATH_PREFIX__` be used here too?
```diff
-match(page.matchPath, browserLoc.pathname)
+match(__PATH_PREFIX__  + page.matchPath, browserLoc.pathname)
```

Reply:
that would be different tho i think? page.matchPath && will bail for non client routes